bliss/surveys/dc2.py

```diff
@@ -293,6 +183,161 @@ def test_dataloader(self):
             self.test_dataset, batch_size=self.batch_size, num_workers=self.num_workers // 2
         )
 
+    def get_plotting_sample(self, image_idx):
+        if self.image_files is None or self.bg_files is None:
+            self._load_image_and_bg_files_list()
+
+        result_dict = load_image_and_catalog(
+            image_idx,
+            self.image_files,
+            self.bg_files,
+            self.image_lim,
+            self.bands,
+            self.n_bands,
+            self.cat_path,
+            self.min_flux_threshold,
+        )
+        return {
+            "tile_catalog": result_dict["tile_dict"],
+            "image": torch.from_numpy(result_dict["inputs"]["image"]),
+            "background": torch.from_numpy(result_dict["inputs"]["bg"]),
+            "match_id": result_dict["other_info"]["match_id"],
+            "full_catalog": result_dict["other_info"]["full_cat"],
+            "wcs": result_dict["other_info"]["wcs"],
+            "psf_params": result_dict["inputs"]["psf_params"],
+        }
+
+
+def squeeze_tile_dict(tile_dict):
+    tile_dict_copy = copy.copy(tile_dict)
+    for k, v in tile_dict_copy.items():
+        if k != "n_sources":
+            tile_dict_copy[k] = rearrange(v, "1 h w nh nw -> h w nh nw")
+    tile_dict_copy["n_sources"] = rearrange(tile_dict_copy["n_sources"], "1 h w -> h w")
+    return tile_dict_copy
+
+
+def unsqueeze_tile_dict(tile_dict):
+    tile_dict_copy = copy.copy(tile_dict)
+    for k, v in tile_dict_copy.items():
+        if k != "n_sources":
+            tile_dict_copy[k] = rearrange(v, "h w nh nw -> 1 h w nh nw")
+    tile_dict_copy["n_sources"] = rearrange(tile_dict_copy["n_sources"], "h w -> 1 h w")
+    return tile_dict_copy
```

Comment:
Is there anyway to avoid squeezing and then unsqueezing these tensors? i.e., Can we just index them according to their original format?

Reply:
OK, got it